src/feature/hs/hs_descriptor.c

```diff
@@ -2325,7 +2338,12 @@ desc_decode_encrypted_v3(const hs_descriptor_t *desc,
   /* Initialize the descriptor's introduction point list before we start
    * decoding. Having 0 intro point is valid. Then decode them all. */
   desc_encrypted_out->intro_points = smartlist_new();
-  decode_intro_points(desc, desc_encrypted_out, message);
+
+  if (decode_intro_points(desc, desc_encrypted_out, message) < 0) {
+    log_warn(LD_REND, "We have gotten duplicate intro points. "
+                      "Rejecting descriptor.");
+    goto err;
+  }
```

Comment:
This entire commit shouldn't be. This is actually allowed by the protocol for now.

What needs to happen is avoid a client connecting to an introduction point that is in the failure cache for the _same_ descriptor.